public/components/security/roles-mapping/helpers/rule-editor.helper.ts

```diff
@@ -11,24 +11,34 @@ export const getJsonFromRule = (internalUserRules, rules, logicalOperator) => {
     tmpRule[item.searchOperation] = {};
     tmpRule[item.searchOperation][item.user_field] = item.value;
     return tmpRule;
-  });
+  }); 
+  
   if (usersRulesArray.length && rulesArray.length) {
-    ruleObject['OR'] = [
-      {
-        OR: usersRulesArray,
-      },
-      {
-        [logicalOperator]: rulesArray,
-      },
-    ];
-  } else {
-    if (usersRulesArray.length) {
-      ruleObject['OR'] = usersRulesArray;
+    if(rulesArray.length == 1){
```

Comment:
this is causing bad behavior because is removing the `userRulesArray`

Reply:
please add some unit test to this function with this uses cases
- Giving userRulesArray NO empty and rulesArray NO empty, should return an object with one `'OR'` array with both arrays as items with all items
- Giving userRulesArray with 1 item  and rulesArray empty, should return the single item
- Giving rulesArray with 1 item  and userRulesArray empty, should return the single item
- Giving rulesArray with 2 o more items  and userRulesArray empty, should return an object with one item equals to the `logicalOperator` (OR or AND) array with all items
 - Giving userRulesArray with 2 o more items  and rulesArray empty, should return an object with one `'OR'` array with all items